src/Files.App/Helpers/Application/AppLanguageHelper.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using System.Globalization;
+using Windows.Globalization;
+
+namespace Files.App.Helpers
+{
+    /// <summary>
+    /// A helper class for managing application languages.
+    /// </summary>
+    public static class AppLanguageHelper
+    {
+		/// <summary>
+		/// A constant string representing the default language ID.
+		/// It is initialized as an empty string.
+		/// </summary>
+		private static readonly string _defaultId = string.Empty;
+
+        /// <summary>
+        /// A collection of available languages.
+        /// </summary>
+        public static ObservableCollection<AppLanguageItem> Languages { get; }
+
+        /// <summary>
+        /// The index of the currently selected language.
+        /// </summary>
+        public static int SelectedIndex { get; private set; }
```

Comment:
The problem occurs when the default value is the same as the one already contained, now it is not possible to determine exactly what index it is, but the helper has determined this number correctly and it is no longer necessary to recalculate the index every time it is needed. I would keep the index.

Reply:
Or I would replace the `string PreferredLanguageCode` with `AppLanguageItem PreferredLanguage` then it would be possible to look up the index, but the code would be faster without having to keep calculating the index